src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -223,7 +224,13 @@ private <T extends SyntaxElement> T parse(Iterator<? extends SyntaxElementInfo<?
 						try {
 							res = parse_i(pattern, 0, 0);
 						} catch (MalformedPatternException e) {
-							throw new RuntimeException("pattern compiling exception, element class: " + info.c.getName(), e);
+							String message = "pattern compiling exception, element class: " + info.c.getName();
+							try {
+								JavaPlugin providingPlugin = JavaPlugin.getProvidingPlugin(info.c);
+								message += " (provided by " + providingPlugin.getName() + ")";
+							} catch (IllegalArgumentException | IllegalStateException ignored) {}
+							throw new RuntimeException(message);
```

Comment:
You removed the exception from the runtime exception constructor.
```suggestion
							throw new RuntimeException(message, e);
```